src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs

```diff
@@ -643,6 +650,42 @@ public async Task<IActionResult> Unpublish(string contentItemId, string returnUr
                 : RedirectToAction(nameof(List));
         }
 
+        public async Task<IActionResult> OwnerFilterUserSearch(string searchTerm, string selectedOwnerUserName, int page = 1)
+        {
+            var pageSize = 50;
+
+            var query = _session.Query<User, UserIndex>();
+
+            if (!string.IsNullOrEmpty(searchTerm))
+            {
+                query.Where(x => x.NormalizedUserName.Contains(_userManager.NormalizeName(searchTerm)));
+            }
+
+            var users = await query
```

Comment:
We may need to cache this for optimization otherwise we'll significantly increase the server load.

Reply:
I had the same thought. It at least isn't page blocking with it loading after page render. I was thinking it would be fine if it would only start loading users when the select dropdown is shown, but bootstrap-select does not make that easy. Thoughts?

Reply:
@douwinga we agree on your suggestion to only load the list on the first drop down, and maybe less than 50, probably something close to what the drop down can fit by default.

Reply:
If "Owner" is indexed, you may also not have to load all users, but just the distinct values of this field.

Reply:
unfortunately, I am running into too many issues with bootstrap-select's new ajax features. Paging is not actually triggering past the 2nd page and it appears the loading of the 2nd page occurs way too early. Also waiting for first drop down is not really doable without getting hacky. Is there another library we could use for this?

Reply:
@douwinga a better idea Sebestien mentioned was to not load any users by default. If the user clicks on the filter menu, then make an AJAX request that would fetch the first 50 users as you are doing today. This way you don't need to cache anything and also you don't need to render anything on every single request.

Reply:
@MikeAlhayek ya, I am just struggling to implement that with bootstrap-select

Reply:
let me know what you think. I made it load the data on first dropdown toggle. It is gross, but I am not sure how else to do it. bootstrap-select needs help, but I want to use it for this so that it is consistent with all the other filter dropdowns,